cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -311,6 +312,34 @@ func (p *dockerProject) packBuild(
 					inDockerOutputPath,
 				))
 		}
+
+		// Support for FastAPI apps since the Oryx builder does not support it yet
+		if svc.Language == ServiceLanguagePython {
+			prj, err := appdetect.DetectDirectory(ctx, svc.Path())
+			if err != nil {
+				return nil, err
+			}
+
+			for _, dep := range prj.Dependencies {
+				if dep == appdetect.PyFastApi {
+					launch, err := appdetect.PyFastApiLaunch(prj.Path)
+					if err != nil {
+						return nil, err
+					}
+
+					// If launch isn't detected, fallback to default Oryx runtime logic, which may recover for scenarios
+					// such as a simple main entrypoint launch.
+					if launch != "" {
+						environ = append(environ,
+							"POST_BUILD_COMMAND=pip install uvicorn",
+							//nolint:lll
+							"ORYX_RUNTIME_SCRIPT=oryx create-script -appPath ./oryx-output -bindPort 80 -userStartupCommand "+
+								"'uvicorn "+launch+" --port $PORT --host $HOST' && ./run.sh")
```

Comment:
We typically recommend gunicorn with uvicorn for production, as it can better handle multiple processes. The FastAPI docs also suggest that, last I checked. Here's the Python team's recommendations for WSGI/ASGI scripts:

https://microsoft-my.sharepoint.com/:w:/p/pamelafox/EZ1t4Z2q5e9Dr_TED-nnf0sBJTdsK9YOMosTmFoEZa6AMA?e=D8crRI

Reply:
When running inside a docker container, with instances being served by multiple containers (as opposed to multiple worker process on a single VM), Uvicorn is sufficient and recommended. The official FastAPI doc on [running in Docker container](https://fastapi.tiangolo.com/deployment/docker/#replication-number-of-processes).

Reply:
Thanks, I guess we misinterpreted the guidance in terms of the best approach for app service. cc @tonybaloney who I've discussed this with in the past.